xamarin/Xamarin.Forms
# Fluent layout support

### Description of Change ###

I prefer code over xaml for creating UI. I presume, for those guys, who like code UI more than XAML, it would be convenient/comfortable to use fluent layout extensions methods.

Let's check on simple sample (we want to create button and label and bind their properties to any view model (without details of VM.. it is abstraction)

NOW:
```csharp
public class TestPage : ContentPage
{
	public TestPage()
	{
		var button = new Button
		{
			Text = "Button"
		};
		button.SetBinding(Button.CommandProperty, "TapCommand");

		var label = new Label();
		label.SetBinding(Label.TextProperty, "LabelText");
		label.SetBinding(Label.TextColorProperty, "LabelColor");

		Content = new StackLayout
		{
			Children =
			{
				button,
				label
			}
		};
	}
}
```

FLUENT:
```csharp
public class TestPage : ContentPage
{
	public TestPage()
 	=> Content = new StackLayout().Children(
		new Button
	 	{
		 	Text = "Button"
	 	}.Command("TapCommand"),
		new Label()
			.Bind(Label.TextProperty, "LabelText")
			.Bind(Label.TextColorProperty, "LabelColor"));
}	
```

You needn't have to create local temp variables for bindings etc.
There are base methods only, but we can improve and extend them easily.

Any ideas?

### API Changes ###
Added new extensions methods

### Platforms Affected ### 
- Core

### Behavioral/Visual Changes ###
None

### Before/After Screenshots ### 
Not applicable

### Testing Procedure ###
Xamarin.Forms.Controls project, App.cs file:
uncomment ```//MainPage = new RootPages.RootFluentPage(); // fluent test ``` and run.
See RootFluentPage class. Layout is fluent
